crates/oxc_linter/src/rules/unicorn/prefer_set_has.rs

```diff
@@ -0,0 +1,900 @@
+use itertools::Itertools;
+use oxc_ast::{
+    ast::{Expression, MemberExpression, VariableDeclarationKind},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::ScopeId;
+use oxc_span::Span;
+use phf::phf_set;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+const ARRAY_METHODS_RETURNS_ARRAY: phf::Set<&'static str> = phf_set! {
+   "concat",
+    "copyWithin",
+    "fill",
+    "filter",
+    "flat",
+    "flatMap",
+    "map",
+    "reverse",
+    "slice",
+    "sort",
+    "splice",
+    "toReversed",
+    "toSorted",
+    "toSpliced",
+    "with",
+};
+
+fn prefer_set_has_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Switch to `Set`")
+        .with_help("should be a `Set`, and use .has()` to check existence or non-existence.")
```

Comment:
These messages are reversed: the `warn()` string should describe the error/issue, the help text should explain how to fix it:

```suggestion
    OxcDiagnostic::warn("Should be a `Set`, and use `.has()` to check existence or non-existence.")
        .with_help("Switch to `Set`")
```